scripts/start-deployPufferfish

```diff
@@ -1,44 +1,43 @@
 #!/bin/bash
 
-. ${SCRIPTS:-/}start-utils
+# shellcheck source=start-utils
+. "${SCRIPTS:-/}start-utils"
+
 set -euo pipefail
 isDebugging && set -x
 
 IFS=$'\n\t'
 
-if [ "${VERSION}" != "LATEST" ] && [ "${VERSION}" != "1.18" ] ; then
-  log "ERROR: Pufferfish server type only supports VERSION=LATEST, VERSION=1.18. Note that these are branches, not #.#.# versions."
+if [ "${VANILLA_VERSION%.*}" != "1.18" ] && [ "${VANILLA_VERSION%.*}" != "1.17" ]; then
+  log "ERROR: Pufferfish server type only supports versions 1.18 or 1.17, use PUFFERFISH_BUILD to select the the correct build 47 => 1.18.1, 50 => 1.18.2 etc"
   exit 1
 fi
 
-: ${PUFFERFISH_BUILD:=lastSuccessfulBuild}
-
-if  [ "${VERSION}" = "LATEST" ] || [ "${VERSION}" = "1.18" ]; then
-  PUFFERFISH_BRANCH="1.18"
-  PUFFERFISH_VERSION="1.18.1-R0.1"
-fi
+: "${PUFFERFISH_BUILD:=lastSuccessfulBuild}"
 
-log "Using Pufferfish-${PUFFERFISH_BRANCH} branch"
+PUFFERFISH_BUILD_JSON=$(curl -X GET -s "https://ci.pufferfish.host/job/Pufferfish-${VANILLA_VERSION%.*}/${PUFFERFISH_BUILD}/api/json")
+# Example: "url": "https://ci.pufferfish.host/job/Pufferfish-1.18/50/",
+PUFFERFISH_BUILD_URL=$(jq -n "$PUFFERFISH_BUILD_JSON" | jq -jc '.url // empty' )
+# Example: "fileName": "pufferfish-paperclip-1.18.2-R0.1-SNAPSHOT-reobf.jar",
+PUFFERFISH_BUILD_FILENAME=$(jq -n "$PUFFERFISH_BUILD_JSON" | jq -jc '.artifacts[].fileName // empty' )
+PUFFERFISH_BUILD_DOWNLOAD_URL="${PUFFERFISH_BUILD_URL}artifact/build/libs/${PUFFERFISH_BUILD_FILENAME}"
 
-export SERVER=pufferfish-${PUFFERFISH_BRANCH}-${PUFFERFISH_BUILD}.jar
 
 log "Removing old Pufferfish versions ..."
 shopt -s nullglob
 for f in pufferfish-*.jar; do
-  [[ $f != $SERVER ]] && rm $f
+  [[ $f != "$PUFFERFISH_BUILD_FILENAME" ]] && rm "$f"
```

Comment:
...why the change from SERVER in the first place?

Reply:
because  I was getting away from the static concatenation of filename. Will pivot file name to SERVER.